server/server.go

```diff
@@ -557,28 +616,55 @@ func (server *Server) startServer(srv *http.Server, globalConfiguration GlobalCo
 	}
 }
 
-func (server *Server) prepareServer(entryPointName string, router *middlewares.HandlerSwitcher, entryPoint *EntryPoint, middlewares ...negroni.Handler) (*http.Server, error) {
-	log.Infof("Preparing server %s %+v", entryPointName, entryPoint)
+func (server *Server) prepareServer(entryPointName string, entryPoint *EntryPoint, router *middlewares.HandlerSwitcher, middlewares ...negroni.Handler) (*http.Server, error) {
+	readTimeout, writeTimeout, idleTimeout := buildServerTimeouts(server.globalConfiguration)
+	log.Infof("Preparing server %s %+v with readTimeout=%s writeTimeout=%s idleTimeout=%s", entryPointName, entryPoint, readTimeout, writeTimeout, idleTimeout)
+
 	// middlewares
-	var negroni = negroni.New()
+	n := negroni.New()
 	for _, middleware := range middlewares {
-		negroni.Use(middleware)
+		n.Use(middleware)
 	}
-	negroni.UseHandler(router)
+	n.UseHandler(router)
+
 	tlsConfig, err := server.createTLSConfig(entryPointName, entryPoint.TLS, router)
 	if err != nil {
 		log.Errorf("Error creating TLS config: %s", err)
 		return nil, err
 	}
 
 	return &http.Server{
-		Addr:        entryPoint.Address,
-		Handler:     negroni,
-		TLSConfig:   tlsConfig,
-		IdleTimeout: time.Duration(server.globalConfiguration.IdleTimeout),
+		Addr:         entryPoint.Address,
+		Handler:      n,
+		TLSConfig:    tlsConfig,
+		ReadTimeout:  readTimeout,
+		WriteTimeout: writeTimeout,
+		IdleTimeout:  idleTimeout,
 	}, nil
 }
 
+func buildServerTimeouts(globalConfig GlobalConfiguration) (readTimeout, writeTimeout, idleTimeout time.Duration) {
+	readTimeout = time.Duration(0)
+	writeTimeout = time.Duration(0)
+	if globalConfig.RespondingTimeouts != nil {
+		readTimeout = time.Duration(globalConfig.RespondingTimeouts.ReadTimeout)
+		writeTimeout = time.Duration(globalConfig.RespondingTimeouts.WriteTimeout)
+	}
+
+	// Backwards compatibility for deprecated IdleTimeout
+	idleTimeout = time.Duration(globalConfig.IdleTimeout)
+	// When deprecated option is not set, set value to DefaultIdleTimeout
+	if idleTimeout == 0 {
+		idleTimeout = time.Duration(DefaultIdleTimeout)
+	}
+	// When RespondingTimeouts.IdleTimout is configured and its not the default value, always use that setting
+	if globalConfig.RespondingTimeouts != nil && time.Duration(globalConfig.RespondingTimeouts.IdleTimeout) != time.Duration(DefaultIdleTimeout) {
+		idleTimeout = time.Duration(globalConfig.RespondingTimeouts.IdleTimeout)
+	}
+
```

Comment:
Thanks for the suggestion, I think its much clearer than before. I just had to tweak the implementation a bit because `globalConfig.IdleTimeout` is a `flaeg.Duration` and can not be compared to `nil`.